src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -466,11 +467,278 @@ type Moon {
+                            },
+                            {
+                              ""action"": ""delete""
+                            }
+                          ]
+                        },
+                        {
+                          ""role"": ""authenticated"",
+                          ""actions"": [
+                            {
+                              ""action"": ""create""
+                            },
+                            {
+                              ""action"": ""read""
+                            },
+                            {
+                              ""action"": ""update""
+                            },
+                            {
+                              ""action"": ""delete""
+                            }
+                          ]
+                        }
+                      ],
+                        ""mappings"": {
+                          """ + entityBackingColumn + @""": """ + entityExposedName + @"""
+                        }
+                      }
+                  }
+              }");
+
+        }
+
+            [TestCleanup]
         public void CleanupAfterEachTest()
         {
             TestHelper.UnsetAllDABEnvironmentVariables();
+            if (File.Exists(HOT_RELOAD_CONFIG_FILE_NAME))
+            {
+                File.Delete(HOT_RELOAD_CONFIG_FILE_NAME);
+            }
+        }
+
+        /// <summary>
+        /// Start DAB with the pre-formed config file, assert that rest and GQL
+        /// requests for the book database object returns the expected results.
+        /// Then hot-reload the config runtime paths both for rest and GQL, and
+        /// again assert that the requests return the expected results against
+        /// the book database object.
+        /// </summary>
+        /// <returns></returns>
+        [TestCategory(MSSQL_ENVIRONMENT)]
+        [TestMethod("Hot-reload runtime paths.")]
+        public async Task HotReloadConfigRuntimePathsEndToEndTest()
+        {
+            // Arrange
+            string restBookContents = $"{{\"value\":{BOOK_DBO_CONTENTS}}}";
+            string restPath = "/restApi";
+            string gQLPath = "/gQLApi";
+
+            string graphQLQueryName = "books";
+            string graphQLQuery = @"{
+                books(first: 100) {
+                    items {
+                        id
+                        title
+                        publisher_id
+                    }
+                }
+            }";
+
+            GenerateConfigFile(connectionString: $"{GetConnectionStringFromEnvironmentConfig(TestCategory.MSSQL)}");
+
+            TestServer server = new(Program.CreateWebHostBuilder(new string[] { "--ConfigFileName", HOT_RELOAD_CONFIG_FILE_NAME }));    
+            HttpClient httpClient = server.CreateClient();
+            RuntimeConfigProvider configProvider = server.Services.GetService<RuntimeConfigProvider>();
```

Comment:
while this was done in the past, it does seem odd that we need to acquire the runtimeconfigprovider from the testserver.Perhaps a modification to `GraphQLRequestExecutor.PostGraphQLRequestAsync(` would make these new tests cleaner:
1. Move most logic of `GraphQLRequestExecutor.PostGraphQLRequestAsync(` to another public static function except for using "runtimeconfigprovider" to get the graphql path, the new public static function would have the same method signature with exception to runtimeconfigprovider, instead of that, you'd just have `string graphQLPath`. then that would enable you to avoid dealing with test server internals. 
1. we wouldn't need to refactor other tests.. yet. They could move to improved api later.

Reply:
I will mention this and some other improvements we can do in the task we have tracking test cleanup and improvements.